--- gnome-session-3.21.3/gnome-session/main.c	2016-05-26 23:04:18.000000000 +0200
+++ gnome-session-3.21.3-1/gnome-session/main.c	2016-07-05 14:23:01.381991025 +0200
@@ -189,7 +189,7 @@
                               TRUE);
 
         /* +2 for our new arguments, +1 for NULL */
-        new_argv = g_malloc (argc + 3 * sizeof (*argv));
+        new_argv = g_malloc ((argc + 3) * sizeof (*argv));
 
         new_argv[0] = "dbus-launch";
         new_argv[1] = "--exit-with-session";
