--- a/util.c	2016-07-09 15:08:05.559942471 +0200
+++ b/util.c	2016-07-09 15:08:16.244942097 +0200
@@ -198,12 +198,14 @@
       xfree (homedir);
       return (0);
     }
-  else if (rl_line_buffer[start] != '~')
+  else if (start >= 0 && rl_line_buffer[start] != '~')
     {
       for (; !whitespace (rl_line_buffer[start]) && start >= 0; start--)
         ;
       start++;
     }
+  else if (start < 0)
+    start = 0;
 
   end = start;
   do
